### ISSUE TITLE
Both cpu and mps?

### ISSUE THREAD

**Issue by ajtanskanen** [brycedrennan/imaginAIry#300]
When I installed imaginAIry and tried to run it, I get the following problem. I am trying to run the package on Apple MacBook M1 Pro, but still imaginAIry tries to use both mps and cpu? What is the issue here?

```bash
imagine --model SD-2.0-depth --init-image 2560px-Ferdinand_von_Wright_-_Taistelevat_metsot_-_1886_-_oljymaalaus_-_Ateneumin_taidemuseo.png  --init-image-strength 0.05  "professional nature shot of two birds fighting of a female bird" -r 4 -w 1024 -h 1024 --steps 50 --log-level DEBUG
```
```bash
✨ Generate images faster using a persistent shell session. Just run `aimg` to start. This makes generation and editing much quicker since the model can stay loaded in memory.

2023-04-09 10:59:05,699 [INFO] imaginairy.cli.shared:95: Received 1 prompt(s) and 1 input image(s). Will repeat the generations 4 times to create 4 images.
2023-04-09 10:59:06,295 [INFO] imaginairy.api:166: Generating 🖼  1/4: "professional nature shot of two birds fighting of a female bird" 1024x1024px negative-prompt:"Ugly, duplication, duplicates, mutilation, deformed, mutilated, mutation, twisted body, disfigured, bad anatomy, out of frame, extra fingers, mutated hands, poorly drawn hands, extra limbs, malformed limbs, missing arms, extra arms, missing legs, extra legs, mutated hands, extra hands, fused fingers, missing fingers, extra fingers, long neck, small head, closed eyes, rolling eyes, weird eyes, smudged face, blurred face, poorly drawn face, mutation, mutilation, cloned face, strange mouth, grainy, blurred, blurry, writing, calligraphy, signature, text, watermark, bad art," seed:934599694 prompt-strength:7.5 steps:50 sampler-type:k_dpmpp_2m init-image-strength:0.05 model:SD-2.0-depth
2023-04-09 10:59:06,606 [INFO] imaginairy.model_manager:126: Loading model /Users/anttitanskanen/.cache/huggingface/hub/models--stabilityai--stable-diffusion-2-depth/snapshots/d41a0687231847e8bd55f43fb1f576afaeefef19/512-depth-ema.ckpt onto mps:0 backend...
  0%|                                                                                                                | 0/12 [00:00<?, ?it/s]
Traceback (most recent call last):
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/bin/imagine", line 7, in <module>
    imaginairy.cli.imagine.imagine_cmd()
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/cli/imagine.py", line 97, in imagine_cmd
    return _imagine_cmd(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/cli/shared.py", line 181, in _imagine_cmd
    filenames = imagine_image_files(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/api.py", line 60, in imagine_image_files
    for result in imagine(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/api.py", line 172, in imagine
    result = _generate_single_image(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/api.py", line 484, in _generate_single_image
    comp_image = _generate_composition_image(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/api.py", line 658, in _generate_composition_image
    result = _generate_single_image(composition_prompt)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/api.py", line 508, in _generate_single_image
    samples = sampler.sample(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/samplers/kdiff.py", line 135, in sample
    samples = self.sampler_func(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/vendored/k_diffusion/sampling.py", line 929, in sample_dpmpp_2m
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/samplers/kdiff.py", line 263, in forward
    noise_pred = get_noise_prediction(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/samplers/base.py", line 81, in get_noise_prediction
    noise_pred_neutral, noise_pred_positive = denoise_func(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/samplers/kdiff.py", line 246, in _wrapper
    return self.inner_model(
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/vendored/k_diffusion/external.py", line 130, in forward
    eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/vendored/k_diffusion/external.py", line 160, in get_eps
    return self.inner_model.apply_model(*args, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/modules/diffusion/ddpm.py", line 1288, in apply_model
    x_recon = self.model(x_noisy, t, **cond)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/Users/anttitanskanen/opt/miniconda3/envs/image/lib/python3.10/site-packages/imaginairy/modules/diffusion/ddpm.py", line 1732, in forward
    xc = torch.cat([x] + c_concat, dim=1)
RuntimeError: torch.cat(): all input tensors must be on the same device. Received mps:0 and cpu
```

**Comment by KixPanganiban** [comment]
I think it's the `SD-2.0-depth` model that's causing this. Try something else like `openjourney-v2` (or remove the `--model` altogether) and it works

**Comment by brycedrennan** [comment]
confirmed

**Comment by brycedrennan** [comment]
fix in just released 12.0.1
